doc/api/n-api.md

```diff
@@ -4247,6 +4247,125 @@ This API represents the invocation of the `ArrayBuffer` `IsDetachedBuffer`
 operation as defined in [Section 24.1.1.2][] of the ECMAScript Language
 Specification.
 
+## Working with JavaScript SharedArrayBuffers
+
+SharedArrayBuffers are used to represent fixed-length binary data buffers
+that can be shared across multiple workers. They are similar to ArrayBuffers
+but allow for concurrent access from multiple threads.
+
+### `napi_is_sharedarraybuffer`
+
+<!-- YAML
+added: REPLACEME
+napiVersion: 10
+-->
+
+```c
+napi_status napi_is_sharedarraybuffer(napi_env env, napi_value value, bool* result)
+```
+
+* `[in] env`: The environment that the API is invoked under.
+* `[in] value`: The JavaScript value to check.
+* `[out] result`: Whether the given `napi_value` represents a `SharedArrayBuffer`.
+
+Returns `napi_ok` if the API succeeded.
+
+This API checks if the Object passed in is a `SharedArrayBuffer`.
+
+### `napi_create_sharedarraybuffer`
+
+<!-- YAML
+added: REPLACEME
+napiVersion: 10
+-->
+
+```c
+napi_status napi_create_sharedarraybuffer(napi_env env,
+                                         size_t byte_length,
+                                         void** data,
+                                         napi_value* result)
+```
+
+* `[in] env`: The environment that the API is invoked under.
+* `[in] byte_length`: The length in bytes of the shared array buffer to create.
+* `[out] data`: Pointer to the underlying byte buffer of the `SharedArrayBuffer`.
+  `data` can optionally be ignored by passing `NULL`.
+* `[out] result`: A `napi_value` representing a JavaScript `SharedArrayBuffer`.
+
+Returns `napi_ok` if the API succeeded.
+
+This API returns a Node-API value corresponding to a JavaScript `SharedArrayBuffer`.
+`SharedArrayBuffer`s are used to represent fixed-length binary data buffers that
+can be shared across multiple workers.
+
+The `SharedArrayBuffer` allocated will have an underlying byte buffer whose size is
+determined by the `byte_length` parameter that's passed in.
+The underlying buffer is optionally returned back to the caller in case the
+caller wants to directly manipulate the buffer. This buffer can only be
+written to directly from native code. To write to this buffer from JavaScript,
+a typed array or `DataView` object would need to be created.
+
+JavaScript `SharedArrayBuffer` objects are described in
+[Section 24.2][] of the ECMAScript Language Specification.
+
+### `napi_get_sharedarraybuffer_info`
+
+<!-- YAML
+added: REPLACEME
+napiVersion: 10
+-->
+
+```c
+napi_status napi_get_sharedarraybuffer_info(napi_env env,
+                                           napi_value sharedarraybuffer,
+                                           void** data,
+                                           size_t* byte_length)
```

Comment:
Should this be a separate function? Arguably, it would be a sensible behavior change would be to allow `napi_get_arraybuffer_info` to work with `SharedArrayBuffer`s, since most consumers of either type will not care about the difference between the two